pkg/stub/config.go

```diff
@@ -231,6 +232,19 @@ func (h *Handler) processError(opcfg *v1.Config, ctype v1.ConfigConditionType, c
 		}
 		status.Status = cstatus
 		status.Message = log
+
+		// most reasons for degraded status are handled in ClusterOperatorStatusDegradedCondition based on
+		// state in the opcfg correlated with other factors; but when the error should directly drive the
+		// reason text, we set the reason here; at this time, that means either API Server errors or
+		// errors accessing the local file system
+		reason := kerrors.ReasonForError(err)
+		switch {
+		case len(reason) > 0 && reason != kapis.StatusReasonUnknown:
+			status.Reason = fmt.Sprintf("APIServer%sError", reason)
```

Comment:
Would it make sense to simply bubble the reason outward? In these cases would the message in `log` be sufficient?

Reply:
> Would it make sense to simply bubble the reason outward? In these cases would the message in `log` be sufficient?

IMO no, a log in the pod log is not sufficient.  This concise nugget in the reason facilitates triage of degraded clusters seen in OTA analysis.  And getting must gather data from customers has proven lengthy / problematic.

This provides us a low cost / low impact means of connecting sample operator degradation to issues in underlying components.